corehq/apps/reports/standard/sms.py

```diff
@@ -324,48 +334,141 @@ def uses_locations(self):
         abbreviate_phone_number = (self.domain in abbreviated_phone_number_domains)
 
         contact_cache = {}
-        message_type_filter = self.get_message_type_filter()
+
+        def filter_by_types(data_):
+            filtered_types = {t.lower() for t in MessageTypeFilter.get_value(self.request, self.domain)}
+            if not filtered_types:
+                return data_
+
+            relevant_workflows = (
+                WORKFLOW_REMINDER,
+                WORKFLOW_KEYWORD,
+                WORKFLOW_BROADCAST,
+                WORKFLOW_CALLBACK,
+                WORKFLOW_DEFAULT,
+                WORKFLOW_PERFORMANCE,
+            )
+            incl_survey = MessageTypeFilter.OPTION_SURVEY.lower() in filtered_types
+            incl_other = MessageTypeFilter.OPTION_OTHER.lower() in filtered_types
+            is_workflow_relevant = Q(workflow__in=relevant_workflows)
+            workflow_filter = Q(is_workflow_relevant & Q(workflow__in=filtered_types))
+            survey_filter = ~Q(xforms_session_couch_id__exact=None)
+            other_filter = ~Q(is_workflow_relevant | survey_filter)
+            # We can chain ANDs together, but not ORs, so we have to do all the ORs at the same time.
+            if incl_survey and incl_other:
+                filters = (workflow_filter | survey_filter | other_filter)
+            elif incl_survey:
+                filters = (workflow_filter | survey_filter)
+            elif incl_other:
+                filters = (workflow_filter | other_filter)
+            else:
+                filters = workflow_filter
+            return data_.filter(filters)
+
+        def order_by_col(data_):
+            col_fields = ['date', 'couch_recipient', 'phone_number', 'direction', 'text']
+            sort_col = self.request_params.get('iSortCol_0')
+            if sort_col is not None and sort_col < len(col_fields):
+                data_ = data_.order_by(col_fields[sort_col])
+            if self.request_params.get('sSortDir_0') == 'desc':
+                data_ = data_.reverse()
+            return data_
+
+        def get_phone_number(phone_number):
+            if abbreviate_phone_number and phone_number is not None:
+                return phone_number[0:7] if phone_number[0] == "+" else phone_number[0:6]
+            return phone_number
+
+        get_direction = lambda d: self._fmt_direction(d)['html']
+
+        def get_timestamp(date_):
+            timestamp = ServerTime(date_).user_time(self.timezone).done()
+            table_cell = self._fmt_timestamp(timestamp)
+            return table_cell['html']
+
+        def get_contact_link(couch_recipient, couch_recipient_doc_type):
+            doc_info = self.get_recipient_info(couch_recipient_doc_type, couch_recipient, contact_cache)
+            table_cell = self._fmt_contact_link(couch_recipient, doc_info)
+            return table_cell['html']
+
```

Comment:
I think it might be cleaner if these methods were defined on the `MessageLogReport` rather than being defined in the `rows` property